udp/client/clientconn.go

```diff
@@ -159,18 +161,22 @@ func (cc *ClientConn) do(req *pool.Message) (*pool.Message, error) {
 	}
 
 	respChan := make(chan *pool.Message, 1)
-	err := cc.tokenHandlerContainer.Insert(token, func(w *ResponseWriter, r *pool.Message) {
-		r.Hijack()
-		select {
-		case respChan <- r:
-		default:
+	if req.Type() == udpMessage.Confirmable {
+		err := cc.tokenHandlerContainer.Insert(token, func(w *ResponseWriter, r *pool.Message) {
+			r.Hijack()
+			select {
+			case respChan <- r:
+			default:
+			}
+		})
+		if err != nil {
+			return nil, fmt.Errorf("cannot add token handler: %w", err)
 		}
-	})
-	if err != nil {
-		return nil, fmt.Errorf("cannot add token handler: %w", err)
+		defer func() { _, _ = cc.tokenHandlerContainer.Pop(token) }()
+	} else {
+		close(respChan)
```

Comment:
The purpose of `do` is to wait for the response message (eg NonConfirmable). With `close(respChan)` message never comes.  Why did you change it?

Reply:
A non-confirmable request doesn't have a response... so there is no point in waiting for it.

Reply:
This is not true: https://datatracker.ietf.org/doc/html/rfc7252#section-5.2.3

Reply:
Hmm... okay, now I am a bit confused... I am using Non-Confirmable messages in terms of Multicast, which (normally?) has no response.

If non-confirmable messages can have a response, how does the client KNOW if it gets a response or not? Or do we need a "WithNoResponseExpected()" Option?

Reply:
There is a way to set a deadline in context so that if the response does not arrive then the error deadline exceeded is returned.

Reply:
I think setting a deadline is a bad choice... because the timing could be hardware (and cpu/network load) dependent. There should be an option that says "I don't care, just send it and come back".